woocommerce/woocommerce-android
# Issue/5624 new stats header

Closes #5624 

⚠️ PR ready to be reviewed but please **do not merge** until @adamzelinski approves the design changes. 

### Description
Adds a new UI for stats **header** on My Store tab. These are the expected designs (iOS based) 

![StatsHeader](https://user-images.githubusercontent.com/2663464/150182490-c25b6a7d-a838-402e-9753-4787637b4e24.png)

A couple things to consider from this new design: 

- Revenue values are not rounded anymore #5653 
- There is no date range bar anymore. Date ranges are now placed inside the scrollable area as part of MyStoreStatsView. This enables removing some classes such as` MyStoreDateRangeView.kt` and unnecessary listener such as `MyStoreStatsListener.kt`
- Removes `dashboard_recency_text` from below the chart as it is not included in the designs anymore.

### Testing instructions
Open My Store Tab and check that chart header values are displayed correctly ad match designs vbzgTKHV9XsyyN5cxxAuhv-fi-1799%3A86756

### Images/gif

https://user-images.githubusercontent.com/2663464/149821776-76a66ebe-3520-4b4a-83af-7aa3024ad8a6.mp4

Some **special cases** to be aware of: 

| Screenshot 1  | Screenshot 2 | Screenshot 3 |
| ------------- | ------------- | ------------- |
| Visitors count is 0, so we can’t calculate conversion rate | For JCP sites, visitor stats are unavailable due to Jetpack, and thus the conversion rate isn’t available  | In the Today tab where hourly visitor stats are unavailable in the API, both visitor and conversion stats are unavailable |
| <img src="https://user-images.githubusercontent.com/2663464/150148870-e9db5936-8c84-453f-ac8e-e90f76842e8e.png" width="200">  | <img src="https://user-images.githubusercontent.com/2663464/150148332-71be8102-1929-4292-959b-9a737315dffb.png" width="200">  | <img src="https://user-images.githubusercontent.com/2663464/150148341-8167c48e-73d3-467d-997e-12174236a9ee.png" width="200"> |



- [X] Release notes have been updated accordinglt in this PR: #5658
